### PR TITLE
summarize and advise commands

### PR DESCRIPTION
**Pull request by hh2010**
addresses https://github.com/riverscuomo/cuomputer/issues/36
tested locally and got it working

### CODE REVIEW

File: bot/on_message/bots/openai_bot.py [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
 import openai
 # from bot.setup.bots import WeezerpediaAPI
 
 from rich import print
 import random
 
+DEFAULT_MESSAGE_LOOKBACK_COUNT = 12
+
+@dataclass(frozen=True)
+class PromptParams:
```
Comment (hh2010) [inline bot/on_message/bots/openai_bot.py:13]: this generalizes the open ai prompt inputs beyond just the discord message class attributes

File: bot/on_message/bots/openai_bot.py [hunk 2]
```diff
@@ -112,63 +128,59 @@ def should_query_weezerpedia_api(self, last_three_messages):
...
-        # Update the session with messages excluding old system messages
-        self.openai_sessions[message.channel.id] = messages_in_this_channel
+            msg for msg in messages_in_this_channel if msg["role"] != "system" or "[INTERNAL]" not in msg["content"]]
 
         # Add the new system message at the beginning
         new_content = [system_message] + messages_in_this_channel
 
-        # # Step 2: Add any context from Weezerpedia API if needed
+        # Replace the channel messages with the cleaned up content
+        self.openai_sessions[prompt_params.channel_id] = new_content
```
Comment (hh2010) [inline bot/on_message/bots/openai_bot.py:148]: i cleaned this up a bit. the new_content is mutable and passed by reference. so modifying it later will modify it here.

side note: there might be a case to use discord bot's channel history function if we think theres any value in the deleted messages.

File: bot/on_message/bots/openai_bot.py [hunk 3]
```diff
@@ -112,63 +128,59 @@ def should_query_weezerpedia_api(self, last_three_messages):
...
-        new_content = self.append_any_attachments(message, new_content)
+        # Append any attachments to the user's message
+        self.append_any_attachments(prompt_params.attachment_urls, new_content)
 
-        # Step 5: Limit the number of messages in the session to 12
-        if len(messages_in_this_channel) > 12:
-            messages_in_this_channel = messages_in_this_channel[-12:]
-
-        # Step 6: Append all the new content to list of messages in this channel
-        messages_in_this_channel.extend(new_content)
```
Comment (hh2010) [inline bot/on_message/bots/openai_bot.py:152]: is this a bug? `new_content` already includes `messages_in_this_channel`. why did you add it again?

**Comment by hh2010** [comment]
could be worth logging all chat history to a database and studying the discussions in a more comprehensive way

**Comment by riverscuomo** [comment]
"shrine": 909450021353705533,

**Review by riverscuomo — APPROVED** [review]
